pkg/apis/storage/types.go

```diff
@@ -672,3 +672,51 @@ type CSIStorageCapacityList struct {
 	// Items is the list of CSIStorageCapacity objects.
 	Items []CSIStorageCapacity
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// VolumeAttributesClass describes the parameters for a class of volume attributes for
+// which PersistentVolumes can be dynamically provisioned.
+//
+// VolumeAttributesClasses are non-namespaced; the name of the volume attributes class
+// according to etcd is in ObjectMeta.Name.
+type VolumeAttributesClass struct {
+	metav1.TypeMeta
+
+	// Standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// +optional
+	metav1.ObjectMeta
+
+	// spec represents the specification of the desired behavior of the VolumeAttributesClass.
+	Spec VolumeAttributesClassSpec
```

Comment:
After reading the KEP-3751 agian, I'm confuse about the definition of the `VolumeAttributesClass`. The reason is that the definition and example of the `VolumeAttributesClass` are not consistent. 

The [definition](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/3751-volume-attributes-class#kubernetes-api) of the `VolumeAttributesClass` is:

```go
type VolumeAttributesClass struct {
  metav1.TypeMeta
  // +optional
  metav1.ObjectMeta

  // Spec defines the behavior of a VolumeAttributesClass.
  // +optional
  Spec VolumeAttributesClassSpec
}

type VolumeAttributesClassSpec struct {
  // Name of the driver
  DriverName string
  // This field is OPTIONAL.This allows the CO to specify the
  // volume attributes class parameters to apply. 
  // These parameters are immutable 
  Parameters map[string]string
}
```

The [example](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/3751-volume-attributes-class#provisioned-create) of the `VolumeAttributesClass` is:

```yaml
apiVersion: storage.k8s.io/v1alpha1
kind: VolumeAttributesClass
metadata:
  name: silver
parameters:
  iops: "500"
  throughput: "50MiB/s"
```

In the example, the `parameters` is not in the `VolumeAttributesClassSpec` and the `driver` is not in the `VolumeAttributesClass`. I'm not sure whether the `driver` is a part of the `VolumeAttributesClass` or a part of the `VolumeAttributesClassSpec`.

Is the example wrong or the definition wrong? If the definition is wrong, I would like to raise a PR to fix it in the KEP.

cc @sunnylovestiramisu @msau42

Reply:
Please use the example as source of truth. 

```
apiVersion: storage.k8s.io/v1alpha1
kind: VolumeAttributesClass
metadata:
  name: silver
parameters:
  iops: "500"
  throughput: "50MiB/s"
```

The spec was for a previous version of the design but since we are making the VAC immutable, we do not need the spec or status.